ld/bootloader.ld

```diff
@@ -30,20 +30,86 @@ SECTIONS {
 	} > FLASH
 
 	.text (READONLY) : ALIGN(word_align) {
+		*(.text*)
 	} > FLASH
 
-	/* TODO: enable this catch-all once everything's mapped */
-	/* .unmapped 0 : { *(*) } */
+	.rodata (READONLY) : ALIGN(word_align) {
+		*(.rodata*)
+	} > FLASH
+
+	.data : ALIGN(word_align) {
+		__data_start__ = .;
+		*(.data*)
+		. = ALIGN(word_align);
+		__data_end__ = .;
+	} > SRAM AT> FLASH
+
+	__data_load__ = LOADADDR(.data);
+
+	.bss (NOLOAD) : ALIGN(word_align) {
+		__bss_start__ = .;
+		*(.bss*)
+		. = ALIGN(word_align);
+		__bss_end__ = .;
+	} > SRAM
+
+	.stack0 (NOLOAD) : ALIGN(word_align) {
+		__stack4_start__ = .;
+		. += LENGTH(SRAM4);
+		. = ALIGN(word_align);
+		__stack4_end__ = .;
+
+	} > SRAM4
+
+	.stack1 (NOLOAD) : ALIGN(word_align) {
+		__stack5_start__ = .;	
+		. += LENGTH(SRAM5);
+		. = ALIGN(word_align);
+		__stack5_end__ = .;
+	} > SRAM5
+
+	.copy : ALIGN(word_align) {
+		__copy_start__ = .;
+		LONG(ADDR(.data))
+		LONG(LOADADDR(.data))
+		LONG(ADDR(.data) + SIZEOF(.data))
+		__copy_end__ = .;
+	} > FLASH
+
+	.zero : ALIGN(word_align) {
+		__zero_start__ = .;
+		LONG(ADDR(.bss))
+		LONG(ADDR(.bss) + SIZEOF(.bss))
+		LONG(ADDR(.stack0))
+		LONG(ADDR(.stack1) + SIZEOF(.stack1))
+		__zero_end__ = .;
+	} > FLASH
+
+	.ARM.attributes 0 : { *(.ARM.attributes) }
+	.comment        0 : { *(.comment)        }
+	.debug_abbrev   0 : { *(.debug_abbrev)   }
+	.debug_aranges  0 : { *(.debug_aranges)  }
+	.debug_frame    0 : { *(.debug_frame)    }
+	.debug_info     0 : { *(.debug_info)     }
+	.debug_line     0 : { *(.debug_line)     }
+	.debug_line_str 0 : { *(.debug_line_str) }
+	.debug_loclists 0 : { *(.debug_loclists) }
+	.debug_rnglists 0 : { *(.debug_rnglists) }
+	.debug_str      0 : { *(.debug_str)      }
+	.shstrtab       0 : { *(.shstrtab)       }
+	.strtab         0 : { *(.strtab)         }
+	.symtab         0 : { *(.symtab)         }
+	*/
+
+	.unmapped 0 : { *(*) }
 }
 
 __msp0_init__  = ORIGIN(SRAM4) + LENGTH(SRAM4);
 __msp0_limit__ = ORIGIN(SRAM4);
 __msp1_init__  = ORIGIN(SRAM5) + LENGTH(SRAM5);
 __msp1_limit__ = ORIGIN(SRAM5);
 
```

Comment:
```suggestion
```
These need to be removed now that you have `.stack0` and `.stack1`. You'll also need to update the vector table so that the initial `msp` value now points to `__stack0_end__`.

Reply:
It might also make sense to rename `__stack{0,1}_{start,end}__` to `__stack{0,1}_{limit,init}__` since the stack starts at the end of the memory region and grows up to the start of the memory region. This might make code easier to read in the future if you do decide to add support for detecting if you've exceeded the stack limit.